app/Http/Controllers/Backend/Social/TwitterController.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace App\Http\Controllers\Backend;
+
+use Abraham\TwitterOAuth\TwitterOAuth;
+use App\Http\Controllers\Controller;
+use App\Models\User;
+use InvalidArgumentException;
+
+abstract class TwitterController extends Controller
+{
+    /**
+     * @param User $user
+     * @return TwitterOAuth
+     * @throws InvalidArgumentException
+     */
+    public static function getApi(User $user): TwitterOAuth {
+        if ($user->socialProfile->twitter_token == null || $user->socialProfile->twitter_tokenSecret == null) {
+            throw new InvalidArgumentException('User is not connected to twitter');
```

Comment:
You should catch this exception and handle it accordingly. Currently this error would be logged which seems to be a bit unnecessary. Also Either the type of the exception or the message should be changed to sth more fitting. e.g. "Empty Tokens were provided for user"